src/server/game/Globals/AreaTriggerDataStore.cpp

```diff
@@ -272,3 +365,27 @@ AreaTriggerDataStore* AreaTriggerDataStore::Instance()
     static AreaTriggerDataStore instance;
     return &instance;
 }
+
+void AreaTriggerDataStore::OnCreateMap(class Map* map)
```

Comment:
You cannot simply preload all objects on map creation, this is not resistant to grid unloads.

You must add guids to a new container in CellObjectGuids (ObjectMgr.h), see `ObjectMgr::AddCreatureToGrid`/`ObjectMgr::RemoveCreatureFromGrid` for an example of what I'm talking about. Then work with `ObjectGridLoader` to get your areatriggers to spawn (api is predefined, must be `AreaTrigger::LoadFromDB(ObjectGuid::LowType spawnId, Map* map, bool addToMap, bool allowDuplicate)`